### PR TITLE
Use our own detection method for Platform#getExecutionType.

### PR DESCRIPTION
**Pull request by kashike**
Forge's method (getEffectiveSide) does not properly detect the current side, and they are not accepting PRs to fix it.

### CODE REVIEW

**Comment by kashike** [comment]
If anyone has a nicer way to handle this, feel free to chime in.

**Comment by Zidane** [comment]
@kashike 

Want to get this merged in. Any more changes?

**Comment by kashike** [comment]
@Zidane no changed, rebased

File: src/main/java/org/spongepowered/mod/SpongeModPlatform.java [hunk 1]
```diff
@@ -69,4 +59,41 @@ private Type switchOn(Side side) {
...
+                || threadName.equals("Server Shutdown Thread")) {
+            return Type.SERVER;
+        } else if (threadName.startsWith("Server Pinger #")
+                || threadName.startsWith("Chunk Batcher ")
+                || threadName.equals("Client Shutdown Thread")
+                || threadName.equals("Realms-connect-task")
+                || threadName.startsWith("Texture Downloader #")
+                || threadName.startsWith("Server Connector #")
+                || threadName.equals("Timer hack thread")
+                || threadName.equals("Twitch authenticator")
```
Comment (gabizou) [inline src/main/java/org/spongepowered/mod/SpongeModPlatform.java]: Isn't Twitch integration gone?

Comment (kashike) [inline src/main/java/org/spongepowered/mod/SpongeModPlatform.java]: This is still 1.8, so it's still present

**Pull request closed without merging**